src/pages/Home/components/SelectThemeSection/ProductList.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState, useRef, useCallback } from 'react';
+import api from '@/api/api';
+import styled from '@emotion/styled';
+import { ProductCard } from './ProductCard';
+import { Typography } from '@/components/common/Typography';
+
+interface ThemeProduct {
+  id: number;
+  name: string;
+  price: { basicPrice: number; sellingPrice: number; discountRate: number; };
+  imageURL: string;
+  brandInfo: { name: string };
+}
+
+export const ProductList: React.FC<{ themeId: number }> = ({ themeId }) => {
+  const [products, setProducts] = useState<ThemeProduct[]>([]);
+  const [cursor, setCursor] = useState(0);
+  const [hasMore, setHasMore] = useState(true);
+  const [loading, setLoading] = useState(false);
+  const loaderRef = useRef<HTMLDivElement | null>(null);
+
+  const fetchPage = useCallback(async () => {
+    if (loading || !hasMore) return;
+    setLoading(true);
+    try {
+      const res = await api.get<{ data: { list: ThemeProduct[]; cursor: number; hasMoreList: boolean } }>(
+        `/api/themes/${themeId}/products`,
+        { params: { cursor, limit: 10 } }
+      );
+      const { list, cursor: nextCursor, hasMoreList } = res.data.data;
+      setProducts(prev => [...prev, ...list]);
+      setCursor(nextCursor);
+      setHasMore(hasMoreList);
+    } catch (err) {
+      console.error('테마별 상품 로드 실패:', err);
+    } finally {
+      setLoading(false);
+    }
+  }, [themeId, cursor, hasMore, loading]);
```

Comment:
[변경 요청] 페이지네이션 요청 관련한 부분은 useFetchThemeProduct 같은 커스텀 훅으로 분리하면 좋을 것 같네요

```suggestion
  const loaderRef = useRef<HTMLDivElement | null>(null);
  const{ products, hasMore, loading } = useFetchThemeProduct
```

Reply:
수정하였습니다.!!